### PR TITLE
Add new features Get Flash ID

### PR DESCRIPTION
**Pull request by SysAlex**
Для удобства написания скриптов, была добавлена функция выводящая ID флэш памяти установленной в панель программатора.
Поддерживаемые типы пямяти: 25x серия (@DIP8, @SOP8, WSON 8).

Параметр командной строки "-g"

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -455,8 +460,21 @@ void action_write(const char *filename, minipro_handle_t *handle, device_t *devi
...
+	unsigned int get_chip_id = minipro_get_chip_id(handle);
+	minipro_end_transaction(handle);
+	printf("0x%02x\n", get_chip_id);
+	exit(1);
+}
+
 int main(int argc, char **argv) {
 	parse_cmdline(argc, argv);
+  if(cmdopts.get_id) {
+		get_id(device_8_pin);
```
Comment (lkundrak) [inline main.c]: Why is 8_pin hardwired here?

File: main.c [hunk 2]
```diff
@@ -455,8 +460,21 @@ void action_write(const char *filename, minipro_handle_t *handle, device_t *devi
...
+	minipro_begin_transaction(handle);
+	unsigned int get_chip_id = minipro_get_chip_id(handle);
+	minipro_end_transaction(handle);
+	printf("0x%02x\n", get_chip_id);
+	exit(1);
+}
+
 int main(int argc, char **argv) {
 	parse_cmdline(argc, argv);
+  if(cmdopts.get_id) {
```
Comment (lkundrak) [inline main.c]: You don't seem to be following the coding style/indentation of the rest of the code.

**Comment by lkundrak** [comment]
I don't understand your cover message and have no idea why is this useful.

The ID is something internal to minipro -- perhaps you may want to make minipro print the suggested device name instead?

**Comment by timofonic** [comment]
@SysAlex @lkundrak @chenz @rozhuk-im @DavidGriffith @vdudouyt Why is this still open?

**Comment by rozhuk-im** [comment]
Chip ID is not MiniPRO TL866xx internal thing.
IMHO there should be implemented some algo to probe different chips in some sequence. 
I do not know safe sequence to probe, probably this is subject to sniff from vendor software.

**Comment by DavidGriffith** [comment]
@timofonic right.  I don't follow @SysAlex's train of thought.  Probing for chip IDs is not a good idea for reasons detailed in #95.  If someone can propose a safe way to probe a chip when you know generally what it is, then maybe this idea can be revisited.  For instance, suppose we have a chip that we know is a 27c256 UV erasable, but we don't know any other details.

**Pull request closed without merging**